client/tests/integration/permissions.rs

```diff
@@ -24,7 +24,7 @@ fn genesis_transactions_are_validated() {
         AccountId::from_str("alice@wonderland").unwrap(),
     );
 
-    let tx_ref = &mut genesis.transactions.last_mut().unwrap().0;
+    let tx_ref = &mut genesis.transactions_mut().last_mut().unwrap().0;
```

Comment:
Yeah, `TestGenesis` returning an `Option` doesn't make sense.

BTW, I am confused by both tests patching constructed `GenesisNetwork`. They are inserting plain instructions into the payload of a signed transaction. By doing so, AFAIK, they invalidate the signature of the transaction being computed using the previous payload. One test checks that Iroha validates genesis by providing an invalid permission token, but Iroha should not accept genesis because of its signatures I guess. Another test checks whether base64 triggers are handled fine. I don't why the second test passes.

Reply:
Created an question:

- https://github.com/hyperledger/iroha/issues/4162